tests/basetest.py

```diff
@@ -317,8 +292,11 @@ def git_repo():
             cmd("git", "commit", "-m", "Initial")
             cmd("git", "tag", "v1.0.0")
             yield root
-    finally:
-        os.chdir(owd)
+        finally:
+            # Make sure to chdir out of temp_dir before closing it, otherwise windows
+            # will freak out. https://github.com/python/cpython/issues/86962
+            os.chdir(owd)
 
 
 needs_git = pytest.mark.skipif(not has_cmd("git"), reason="git not present on system")
+needs_nix = pytest.mark.skipif(not has_cmd("which", "bash"), reason="not running in a unix-like environment")
```

Comment:
I added the `needs_nix` decorator so that tests that don't need cygwin can still be run on windows.